### PR TITLE
Issues #1513 - Make sure insecure flag works for remote Kustomize bases

### LINKED ISSUE #1513: Introduce `--insecure-skip-tls-verify` option for self-signed HTTP git URLs
**Describe the bug**
It is impossible to connect repositories using HTTP git url if git server uses self-signed certificate.

**To Reproduce**
1. Try to connect repo using `argocd repo add https://<url> --username <username> --password <password> --insecure-ignore-host-key`
2. The connection fails if server uses self signed cert: `x509: cannot validate certificate for <host> because it doesn't contain any IP SANs`

**Expected behavior**
Repository should be successfully connected.

### PR DESCRIPTION
**Pull request by alexmt**
PR implements two following changes:

E2E tests are reworked to use locally running git service instead of private gitlab.com repos ( see test/fixture/testrepos/Procfile )

Ensures that git related env variables are passed to all config generated related commands to support Kustomize remote bases

Closes #1513

### CODE REVIEW

**Comment by alexmt** [comment]
@jannfis , you might be interested in this change. After this change env variables like `GIT_SSL_NO_VERIFY` and `GIT_SSL_CAINFO` are used during manifest generation. As a result it is possible to use kustomize package which remote base uses self signed cert of requires custom known host.

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -896,6 +900,35 @@ type Repository struct {
 	Insecure bool `json:"insecure,omitempty" protobuf:"bytes,7,opt,name=insecure"`
 }
 
+func getCAPath(repoURL string) string {
+	if git.IsHTTPSURL(repoURL) {
+		if parsedURL, err := url.Parse(repoURL); err == nil {
+			if caPath, err := cert.GetCertBundlePathForRepository(parsedURL.Host); err != nil {
```
Comment (jessesuen) [inline pkg/apis/application/v1alpha1/types.go]: We don't care or log about this error?

**Review by jessesuen — APPROVED** [review]

File: pkg/apis/application/v1alpha1/types.go (hunk 1 above)
Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: good point. Warning should help troubleshoting misconfigured cert. Fixed

**Comment by jannfis** [comment]
Awesome, @alexmt - I'm already testing it locally. It's also fantastic that there's now a local Git repository available for E2E tests, instead of relying on Git services on the Internet.

**Pull request merged**